File: Fenicia.Module.Basic/Domains/Order/Commands/CreateOrderCommand.cs
using Fenicia.Common.Enums.Auth;
using Fenicia.Common.Enums.Basic;

namespace Fenicia.Module.Basic.Domains.Order.Commands;

/// <summary>
///     Command to create a new product order.
/// </summary>
public record CreateOrderCommand(
    Guid UserId,
    Guid CustomerId,
    DateTime SaleDate,
    OrderStatus Status,
    List<OrderDetailCommand> Details,
    PaymentMethod PaymentMethod,
    Guid? EmployeeId = null,
    string? Notes = null,
    decimal DiscountAmount = 0);

File: fenicia-web/src/types/basic/order/create-order-command.ts
import { PaymentMethod } from "./payment-method";


export type CreateOrderCommand = {
    customerId: string;
    saleDate: string;
    details: Array<{
        productId: string;
        quantity: number;
        price: number;
    }>;
    discountAmount?: number;
    paymentMethod: PaymentMethod;
    notes?: string;
    employeeId?: string;
    status: string;
};
